platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -303,15 +311,24 @@ HttpResponseWrapper getHttpResponse(String responseConsumerURL, String soapRespo
                     if (request.getFollowRedirects()
                             && HttpStatusCodes.isRedirect(response.getStatusCode())
                             && redirectLocation != null) {
+                        String method = (String) message.get(Message.HTTP_REQUEST_METHOD);
+                        HttpContent content = null;
+                        if (!isRedirectable(method)) {
+                            ByteArrayOutputStream byteArrayOutputStream =
+                                    new ByteArrayOutputStream();
+                            message.setContent(OutputStream.class, byteArrayOutputStream);
+                            BodyWriter bodyWriter = new BodyWriter();
+                            bodyWriter.handleMessage(message);
+                            content =
+                                    new InputStreamContent((String) message.get(Message.CONTENT_TYPE),
+                                            new ByteArrayInputStream(byteArrayOutputStream.toByteArray()));
+                        }
+
                         // resolve the redirect location relative to the current location
                         request.setUrl(new GenericUrl(request.getUrl()
                                 .toURL(redirectLocation)));
-                        // on 303 change method to GET
```

Comment:
It will still hit this code if it receives a 303, but it will reuse whatever method we were trying originally.